lib/Sema/CodeSynthesis.cpp

```diff
@@ -1693,28 +1694,65 @@ ConstructorDecl *swift::createImplicitConstructor(TypeChecker &tc,
       // If var is a lazy property, its value is provided for the underlying
       // storage.  We thus take an optional of the properties type.  We only
       // need to do this because the implicit constructor is added before all
-      // the properties are type checked.  Perhaps init() synth should be moved
+      // the properties are type checked. Perhaps init() synth should be moved
       // later.
       if (var->getAttrs().hasAttribute<LazyAttr>())
         varInterfaceType = OptionalType::get(varInterfaceType);
 
       // Create the parameter.
-      auto *arg = new (context)
+      auto *arg = new (C)
           ParamDecl(VarDecl::Specifier::Default, SourceLoc(), Loc,
                     var->getName(), Loc, var->getName(), decl);
       arg->setInterfaceType(varInterfaceType);
       arg->setImplicit();
       
+      // If this is a var that has a default value, and is only binding one var,
```

Comment:
This method is getting a bit long -- do you mind splitting out the default argument logic?